larva/src/main/java/nl/nn/adapterframework/testtool/SenderThread.java

```diff
@@ -34,22 +39,26 @@
 		log.debug("Request: " + request);
     }
 
-	SenderThread(ISenderWithParameters senderWithParameters, String request, ParameterResolutionContext parameterResolutionContext, boolean convertExceptionToMessage) {
+	SenderThread(ISenderWithParameters senderWithParameters, String request, IPipeLineSession session, boolean convertExceptionToMessage) {
 		name = senderWithParameters.getName();
 		this.senderWithParameters = senderWithParameters;
-		this.parameterResolutionContext = parameterResolutionContext;
 		this.request = request;
+		this.session = session;
 		this.convertExceptionToMessage = convertExceptionToMessage;
 		log.debug("Creating SenderThread for ISenderWithParameters '" + name + "'");
 		log.debug("Request: " + request);
 	}
 
     public void run() {
         try {
+        	if (session==null) {
+        		session = new PipeLineSessionBase();
+        	}
+        	session.put(IPipeLineSession.businessCorrelationIdKey, TestTool.TESTTOOL_CORRELATIONID);
         	if (senderWithParameters == null) {
-				response = sender.sendMessage(TestTool.TESTTOOL_CORRELATIONID, request);
+				response = sender.sendMessage(new Message(request), session).asString();
         	} else {
-				response = senderWithParameters.sendMessage(TestTool.TESTTOOL_CORRELATIONID, request, parameterResolutionContext);
+				response = senderWithParameters.sendMessage(new Message(request),  session).asString();
```

Comment:
Dit moet opgeschoond worden. Je doet 2x het zelfde, ook al is de session null

Reply:
fixed